src/Trigger/IRabbitMQModel.cs

```diff
@@ -26,7 +27,7 @@ public interface IRabbitMQModel
 
         void BasicReject(ulong deliveryTag, bool requeue);
 
-        void BasicPublish(string exchange, string routingKey, IBasicProperties basicProperties, byte[] body);
+        void BasicPublish(string exchange, string routingKey, IBasicProperties basicProperties, ReadOnlyMemory<byte> body);

```

Comment:
[This interface is public](https://github.com/Azure/azure-functions-rabbitmq-extension/blob/dev/src/Trigger/IRabbitMQModel.cs#L9). So, changing the method signature will be a breaking change for anyone using this API. Do you know who uses this?

Do we need this change? If so, should we do it in a backward compatible way? (Introduce a new overload and mark the old one as obsolete).

Reply:
There's no public class that accepts `IRabbitMQModel` as its method argument or has `IRabbitMQModel` as return type. The only path where I could find the model definition leaking was via `IRabbitMQModel` <- `IRabbitMQService` <-`IRabbitMQServiceFactory` but it stops at that point. There are no public classes taking or returning either `IRabbitMQService` or `IRabbitMQServiceFactory` or any of their implementation classes. I guess customer would not be impacted if they cannot work with `IRabbitMQModel` in connection with any other classes in RabbitMQ extension. I will fix the visibility of the classes and interfaces in a future PR.